$OpenBSD: patch-configure,v 1.6 2011/10/13 06:52:49 jasper Exp $

GNOME Bug 627126 - gsettings schema files don't get installed on FreeBSD
glib commit 136e705e8383ff8848e425ac01278102d7badc52

Add amd64 to x86_64 host_cpu.

Don't force the use of -O2, the ports infrastructure
takes care of that

--- configure.orig	Tue Sep 27 12:41:23 2011
+++ configure	Fri Oct  7 10:03:03 2011
@@ -19061,22 +19061,10 @@ install-data-am: install-gsettings-schemas
 
 .SECONDARY: $(gsettings_SCHEMAS)
 
-gsettings__base_list = \
-  sed "$$!N;$$!N;$$!N;$$!N;$$!N;$$!N;$$!N;s/\n/ /g" | \
-  sed "$$!N;$$!N;$$!N;$$!N;s/\n/ /g"
-
-install-gsettings-schemas: $(gsettings_SCHEMAS:.xml=.valid) $(gsettings__enum_file)
+install-gsettings-schemas: $(gsettings_SCHEMAS) $(gsettings__enum_file)
 	@$(NORMAL_INSTALL)
 	test -z "$(gsettingsschemadir)" || $(MKDIR_P) "$(DESTDIR)$(gsettingsschemadir)"
-	@list='\''$(gsettings__enum_file) $(gsettings_SCHEMAS)'\''; test -n "$(gsettingsschemadir)" || list=; \
-	for p in $$list; do \
-	  if test -f "$$p"; then d=; else d="$(srcdir)/"; fi; \
-	  echo "$$d$$p"; \
-	done | $(gsettings__base_list) | \
-	while read files; do \
-	  echo " $(INSTALL_DATA) $$files '\''$(DESTDIR)$(gsettingsschemadir)'\''"; \
-	  $(INSTALL_DATA) $$files "$(DESTDIR)$(gsettingsschemadir)" || exit $$?; \
-	done
+	$(INSTALL_DATA) $^ "$(DESTDIR)$(gsettingsschemadir)"
 	test -n "$(GSETTINGS_DISABLE_SCHEMAS_COMPILE)$(DESTDIR)" || $(GLIB_COMPILE_SCHEMAS) $(gsettingsschemadir)
 
 uninstall-gsettings-schemas:
@@ -21029,7 +21017,7 @@ $as_echo "#define ENABLE_YARR 1" >>confdefs.h
 $as_echo "#define ENABLE_YARR_JIT 1" >>confdefs.h
 
         ;;
-        i*86|x86_64)
+        i*86|x86_64|amd64)
 
 $as_echo "#define ENABLE_JIT 1" >>confdefs.h
 
@@ -21054,7 +21042,7 @@ $as_echo "#define ENABLE_JIT_OPTIMIZE_ARITHMETIC 1" >>
 $as_echo "#define WTF_USE_JIT_STUB_ARGUMENT_VA_LIST 1" >>confdefs.h
 
                 ;;
-                x86_64)
+                x86_64|amd64)
 
 $as_echo "#define WTF_USE_JIT_STUB_ARGUMENT_REGISTER 1" >>confdefs.h
 
@@ -21346,8 +21334,8 @@ fi
 
 # Add the appropriate 'O' level for optimized builds
 if test "$enable_optimizations" = "yes"; then
-   CXXFLAGS="$CXXFLAGS -O2"
-   CFLAGS="$CFLAGS -O2"
+   CXXFLAGS="$CXXFLAGS"
+   CFLAGS="$CFLAGS"
 else
    CXXFLAGS="$CXXFLAGS -O0"
    CFLAGS="$CFLAGS -O0"
